File: python3.9/cookiecutter-aws-sam-quick-start-web-with-connectors/{{cookiecutter.project_name}}/src/handlers/get_all_items.py
import json
import os
import boto3

client = boto3.client('dynamodb')

def getAllItemsHandler(event, context):
    if event["httpMethod"] != "GET":
        raise Exception(f"getAllItems only accept GET method, you tried: {event.httpMethod}")

    data = client.scan(TableName=os.environ["SAMPLE_TABLE"])
    items = data["Items"]
    response = {
        "statusCode": 200,
        "body": json.dumps(items)
    }

    return response

File: ruby/web/{{cookiecutter.project_name}}/src/handlers/get_all_items.rb
# frozen_string_literal: true

require "aws-sdk-dynamodb"

def handler(event:, context:)
  data = client.scan(table_name: ENV["SAMPLE_TABLE"])
  items = data.items

  { statusCode: 200, body: items.to_json }
end

def client
  @client ||= Aws::DynamoDB::Client.new
end
